app/controllers/home_pages_controller.rb

```diff
@@ -1,9 +1,16 @@
 class HomePagesController < ApplicationController
+  before_action :load_product, only: :index
   def index
-    products = Book.newest
-    @pagy, @new_products = pagy products, items: Settings.page_items
+    @pagy, @new_products = pagy @products, items: Settings.page_items
     return @prices = @new_products.pluck(:price) unless params[:locale] == "vi"
 
     @prices = Book.dollar_to_vnd(@new_products)
   end
+
+  private
+  def load_product
+    return @products = Book.search(params[:search]) if params[:search]
+
+    @products = Book.newest
```

Comment:
```suggestion
    @products = Book.search(params[:search]).newest
```
Xử lý parám em để ở scope đi em nhé

Reply:
mà chị thấy scope có handle rồi mà